src/event.js

```diff
@@ -633,7 +633,19 @@ jQuery.each( {
 
 		// Add which for click: 1 === left; 2 === middle; 3 === right
 		if ( !event.which && button !== undefined && rmouseEvent.test( event.type ) ) {
-			return ( button & 1 ? 1 : ( button & 2 ? 3 : ( button & 4 ? 2 : 0 ) ) );
+			if ( button & 1 ) {
+				return 1;
+			}
+
+			if ( button & 2 ) {
+				return 3;
+			}
+
+			if ( button & 4 ) {
+				return 2;
+			}
+
+			return 0;
```

Comment:
Also equivalent, and probably smaller:
```js
return (button & 1) ||
	(button & 2 ? 3 :
	(button & 4) / 2);
```
(where the last `/ 2` can be anything that maps 0 to itself and 4 to 2—`&& 2`, `>> 1`, etc.).